Assets/Scripts/UI/HUD/PauseButtonHandler.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.SceneManagement;
+
+public class PauseButtonHandler : MonoBehaviour
+{
+    public Button pauseButton;
+    public GameObject pauseScreen;
+    public float fadeDuration = 0.5f;
+    public Button exitGameButton;
+    public Image fadeOverlay;
+
+    private bool _isPaused = false;
+    private CanvasGroup _pauseScreenCanvasGroup;
+
+    private void Start()
+    {
+        pauseButton.onClick.AddListener(TogglePause);
+        exitGameButton.onClick.AddListener(ExitGame);
+        _pauseScreenCanvasGroup = pauseScreen.GetComponent<CanvasGroup>();
+        pauseScreen.SetActive(false);
+    }
+
+    private void TogglePause()
+    {
+        _isPaused = !_isPaused;
+        Time.timeScale = _isPaused ? 0f : 1f;
```

Comment:
There seem to be some quirks with TIme.timeScale pausing clients when pausing on the host, but we ca deal with them later